### PR TITLE
Add event display endpoint

### PR DESCRIPTION
**Pull request by eeeschwartz**
## A new use case

This is a step towards a use case we've found useful in the city of Lexington:

> The event is a polygon that I subscribe to

Our example is for notifications when leaf collection crews visit your neighborhood. Here's a [GIF of the Citygram UI for this dataset](https://cloud.githubusercontent.com/assets/428684/10350209/abaf2fd4-6d10-11e5-8c34-e34b7054c25c.gif).

Thanks to Citygram's backend flexibility, making this work is minor UI issue. Citygram triggers notifications equally well for any intersection: 

* **normal Citygram case**: an event point inside of my subscription polygon
* **leaf collection case**: my subscription point (my house) inside of an event polygon

## The proposed change

This PR irons out a kink in the way Citygram displays events during sign-up:

* Currently, it shows the last 7 days of events
* In this case, we want to show the single most recent event for a given polygon no matter when it happened

Rather than change the way the Citygram works on the server-side, this PR let's a given publisher fetch events from a specified endpoint if desired. In our case, a [citygram-services endpoint](https://lexgram-services.herokuapp.com/lexington-leaf-collection-citygram-events-format) with a [slightly different format](https://github.com/citygram/citygram-services/commit/1b5b39bba87ce16122508c8aa1fc219ad319c113#diff-a04a99d91540e24a7bcd9d5838748044R51) works perfectly.

A side perk is that this would allow a standard publisher to display more than 7 days of events or change the wording of the Leaflet pop-up during sign-up, for example.

Any thoughts welcome.

### CODE REVIEW

File: app/assets/js/scripts.js [hunk 1]
```diff
@@ -182,6 +176,23 @@ app.hookupSteps = function() {
 
 };
 
+app.setPublisher = function($publisher) {
+  app.state.publisher_id = $publisher.data('publisher-id');
+
+  // todo: make eventsArePolygons a property of a publisher
+  app.eventsArePolygons = $publisher.data('publisher-title').match(/Leaf Collection/);
```
Comment (bigfleet) [inline app/assets/js/scripts.js]: Is there a way to make this matching less "magical"?  Leaf Collection might not be the only polygon point of interest.  Is this something that could/should be moved to the Publisher model?

Comment (eeeschwartz) [inline app/assets/js/scripts.js]: @bigfleet definitely. The magical matching is a weirdbeard holdover from a polygon proof of concept last year.

I'll put something together for this PR that works more generally.

**Comment by eeeschwartz** [comment]
@bigfleet [publisher.events_are_polygons added](https://github.com/codeforamerica/citygram/commit/ee8af986b9214eb5592ee79d1039903dccda1a51)

**Comment by bigfleet** [comment]
Without totally having thought about this, is it related at all to #212 ?

**Comment by bigfleet** [comment]
@eeeschwartz This looks good, I'm going to evaluate on my laptop before merging.  If that checks out, I'll get it deployed along with bringing Tulsa online.

**Comment by eeeschwartz** [comment]
Thanks @bigfleet!

If you want to test against the production leaf collection dataset, you could:

* Merge and deploy the citygram-services [CORS PR](https://github.com/citygram/citygram-services/pull/42)
* Run the following in IRB:

``
Publisher.where(city: 'Lexington', title: 'Leaf Collection').first.update_all(events_are_polygons:true, event_display_endpoint: 'https://citygram-services.herokuapp.com/lexington-leaf-collection-citygram-events-format')
``

**Comment by eeeschwartz** [comment]
@bigfleet I fixed the conflict in `script.js`. Holler if anything else I can take a look at.

**Pull request merged**